AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/src/Keyrings/RawRSAKeyring.dfy

```diff
@@ -336,7 +364,7 @@ module RawRSAKeyring {
           output.Failure?
     {
       :- Need(
-        this.privateKey.Some? && |this.privateKey.Extract()| > 0,
+        privateKeyMaterial.Some?,
```

Comment:
I see OnEncrypt uses something like
```suggestion
        privateKeyMaterial.Some? && privateKey.Some?,
```
Any reason not to align the two checks?

Reply:
We don't store the private key anymore.
We should change the other one to be only publicKeyMaterial.Some? because we know 
publicKey.None? ==> publicKeyMaterial.None?